src/DynamoCore/Graph/Workspaces/WorkspaceModel.cs

```diff
@@ -880,33 +879,15 @@ public virtual void Clear()
         }
 
         /// <summary>
-        ///     Save to a specific file path, if the path is null or empty, does nothing.
-        ///     If successful, the CurrentWorkspace.FilePath field is updated as a side effect
+        /// Internal save logic exception for serialization.
         /// </summary>
         /// <param name="newPath">The path to save to</param>
-        /// <param name="runtimeCore">The <see cref="ProtoCore.RuntimeCore"/> object
-        /// to obtain serialized trace data for node list to save.</param>
         /// <param name="isBackup">Indicates whether saved workspace is backup or not. If it's not backup,
-        /// we should add it to recent files. Otherwise leave it.</param>
-        public virtual bool SaveAs(string newPath, ProtoCore.RuntimeCore runtimeCore, bool isBackup = false)
+        public virtual bool Save(string newPath, bool isBackup = false)
```

Comment:
Thanks and you are totally right, I think FileName does not need to be changed for backups